# Nama : A. Fuad Ahsan Basir
# NIM : H071201076

# Latihan (1)
Silahkan Download sebuah dataset menggunakan API Kaggle

In [2]:
#Latihan (1)
!pip install kaggle

  Created wheel for kaggle: filename=kaggle-1.5.12-py3-none-any.whl size=73053 sha256=dff75b7482f6ba166603a9e258aa40121ea07b611c70bf7bffc74490a9c5e531
  Stored in directory: c:\users\lenovo\appdata\local\pip\cache\wheels\29\da\11\144cc25aebdaeb4931b231e25fd34b394e6a5725cbb2f50106
Successfully built kaggle


In [12]:
!kaggle datasets list -s Iris

ref                                                        title                                              size  lastUpdated          downloadCount  voteCount  usabilityRating  
---------------------------------------------------------  ------------------------------------------------  -----  -------------------  -------------  ---------  ---------------  
uciml/iris                                                 Iris Species                                        4KB  2016-09-27 07:38:05         308871       3061  0.7941176        
arshid/iris-flower-dataset                                 Iris Flower Dataset                               1010B  2018-03-22 15:18:06          64044        494  0.8235294        
vikrishnan/iris-dataset                                    Iris Dataset                                       999B  2017-08-03 16:00:44           4849         36  0.7647059        
chuckyin/iris-datasets                                     Iris datasets                       

In [14]:
!kaggle datasets download -d  uciml/iris --unzip


  0%|          | 0.00/3.60k [00:00<?, ?B/s]
100%|##########| 3.60k/3.60k [00:00<00:00, 3.61MB/s]


### PENGGUNAAN LIBRARY PANDAS dan NUMPY

Pada materi ini, peserta sudah mendapatkan pemahaman mengenai data dan dataset. Penggunaan library pada Python memberikan kemudahan dalam proses data understanding. Beberapa library yang digunakan adalah library Pandas dan Numpy.

# Latihan (2)

Lakukan import Library Pandas dan Library Numpy

In [1]:
#Latihan(2)
#Import Library Pandas

import pandas as pd

#Import Library Numpy

import numpy as np

### DATAFRAME

DataFrame adalah struktur data 2 dimensi yang berbentuk tabular (mempunyai baris dan kolom). Hampir semua data tidak hanya memiliki 1 kolom tetapi lebih dari 1 kolom, sehingga lebih cocok menggunakan pandas DataFrame untuk mengolahnya.

Penggunaan dataframe pada Python dengan menggunakan syntaks: df.

# Latihan (3)

Panggil file (load dataset) dengan format .csv untuk dataset mengenai bunga Iris yang sudah peserta unduh dari Kaggle, dan akan disimpan di dalam dataframe df. Lalu tampilkan 5 baris awal dataset dengan function head()

In [2]:
#latihan(3)
#Panggil file (load file bernama Iris.csv) dan simpan dalam dataframe Lalu tampilkan 5 baris awal dataset dengan function head()

data = pd.read_csv('Iris.csv')
data.head()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa

# Telaah Data

Pada telaah data, dapat dilakukan untuk mengetahui:
- tipe data dari setiap kolom
- deskripsi statistik data

# Latihan (4)

Tampilkan tipe data dari kolom yang ada pada dataset

In [3]:
#latihan(4)
#Tampilkan tipe data dari kolom yang ada pada dataset

data.dtypes

Id                 int64
SepalLengthCm    float64
SepalWidthCm     float64
PetalLengthCm    float64
PetalWidthCm     float64
Species           object
dtype: object

# Latihan (5)

Apakah tipe Data dari kolom berikut ini: (silakan diisi pada cell di bawah ini)


#Latihan (5)
#Tipe Data dari kolom yang ada di dataset

#### Kolom  "Id" memiliki tipe data =  integer
#### Kolom "SepalLengthCm" memiliki tipe data =  float
#### Kolom "Species" memiliki tipe data =  object

# Latihan (6)

Hitunglah ukuran (jumlah baris dan kolom) dari dataset. Dengan menggunakan method function

In [5]:
#Latihan (6)
#Hitung ukuran (jumlah baris dan kolom) dari dataset

data.shape

(150, 6)

# Latihan (7)

Berapakah jumlah baris, dan jumlah kolom pada dataset? (silakan diisi pada cell di bawah ini)

#Latihan (7)

#### Jumlah Baris pada dataset adalah =  <150>

#### Jumlah kolom pada dataset adalah =  <5>

# Latihan (8)

Tampilkan data yang hanya berisi kolom "Id" dan kolom "Species" dalam bentuk dataframe.

In [26]:
#Latihan (8)
#Tampilkan data untuk kolom "Id" dan kolom "Species" dalam bentuk dataframe

data[['Id', 'Species']]

Id         Species
0      1     Iris-setosa
1      2     Iris-setosa
2      3     Iris-setosa
3      4     Iris-setosa
4      5     Iris-setosa
..   ...             ...
145  146  Iris-virginica
146  147  Iris-virginica
147  148  Iris-virginica
148  149  Iris-virginica
149  150  Iris-virginica

[150 rows x 2 columns]

# Latihan (9)

Tampilkan data dengan dataframe, dan data yang ditampilkan adalah data pada baris dengan indeks 0 (nol) sampai dengan indeks 9 (sembilan)

In [35]:
#Latihan (9)
#Tampilkan data dengan dataframe, dan data yang ditampilkan adalah baris dengan indeks 0 (nol) sampai dengan indeks 9 (sembilan)

data.iloc[:10]

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa
5   6            5.4           3.9            1.7           0.4  Iris-setosa
6   7            4.6           3.4            1.4           0.3  Iris-setosa
7   8            5.0           3.4            1.5           0.2  Iris-setosa
8   9            4.4           2.9            1.4           0.2  Iris-setosa
9  10            4.9           3.1            1.5           0.1  Iris-setosa

# Latihan (10)

Tampilkan data hanya kolom "Id" dan kolom "Species" dengan dataframe, dan yang ditampilkan adalah data pada baris dengan indeks 11 (sebelas) sampai dengan indeks 15 (limabelas)

In [36]:
#Latihan (10)
#Tampilkan data hanya kolom "Id" dan kolom "Species", pada baris dengan indeks 0 (nol) sampai dengan indeks 9 (sembilan)

data[['Id', 'Species']].iloc[11:16]

Id      Species
11  12  Iris-setosa
12  13  Iris-setosa
13  14  Iris-setosa
14  15  Iris-setosa
15  16  Iris-setosa

# Latihan (11)

Pada DataFrame dapat menampilkan beberapa baris pertama/terakhir dari dataset yang di load. Gunakan Method head() dan tail().

Latihan: Tampilkan data pada 8 (delapan) baris pertama dari dataset, dengan dataframe.

In [37]:
#Latihan (11)
#Tampilkan data pada 8 (delapan) baris pertama dari dataset, dengan dataframe

data.head(8)

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa
5   6            5.4           3.9            1.7           0.4  Iris-setosa
6   7            4.6           3.4            1.4           0.3  Iris-setosa
7   8            5.0           3.4            1.5           0.2  Iris-setosa

# Latihan (12)

Tampilkan data pada 3 (tiga) baris terakhir dari dataset, dengan dataframe.

In [38]:
#Latihan (12)
#Tampilkan data pada 3 (tiga) baris terakhir dari dataset, dengan dataframe

data.tail(3)

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  \
147  148            6.5           3.0            5.2           2.0   
148  149            6.2           3.4            5.4           2.3   
149  150            5.9           3.0            5.1           1.8   

            Species  
147  Iris-virginica  
148  Iris-virginica  
149  Iris-virginica

# Deskripsi Statistik Data

DataFrame method describe() menampilkan statistik dasar setiap kolom data yang bertipe numerik, mencakup banyaknya data (count), rerata aritmetik (mean), simpangan baku (std), nilai terkecil (min), kuartil pertama (25%), kuartil kedua/median (50%), kuartil ketiga (75%), dan nilai terbesar (max).

# Latihan (13)

Hitung korelasi dari dataset. Dengan menggunakan method function

In [31]:
#Latihan (13)
#Hitung korelasi dataset

data.corr(method='pearson')

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  \
Id             1.000000       0.716676     -0.397729       0.882747   
SepalLengthCm  0.716676       1.000000     -0.109369       0.871754   
SepalWidthCm  -0.397729      -0.109369      1.000000      -0.420516   
PetalLengthCm  0.882747       0.871754     -0.420516       1.000000   
PetalWidthCm   0.899759       0.817954     -0.356544       0.962757   

               PetalWidthCm  
Id                 0.899759  
SepalLengthCm      0.817954  
SepalWidthCm      -0.356544  
PetalLengthCm      0.962757  
PetalWidthCm       1.000000

# Latihan (14)

Berdasarkan pada perhitungan korelasi di Latihan (11), apakah yang dapat Bapak/Ibu simpulkan sementara? Silakan tuliskan simpulan sementara Bapak/Ibu pada cell di bawah ini.

#latihan (14)
#Simpulan Sementara Hasil Korelasi di latihan (13) 

##### Data yang memiliki nilai korelasi yang mendekati 1 atau -1 berarti data tersebut memiliki hubungan yang hampir sempurna. Berbeda dengan data yang memiliki korelasi hampir sama dengan 0 yang berarti hampir tidak ada keterkaitan antara keduanya.
##### Dari daftar nilai korelasi di atas, dapat dilihat bahwa kolom SepalLengthCm dan SepalWidthCm memilliki nilai korelasi terkecil dibanding nilai korelasi kolom lainnya yaitu -0.109369 yang berarti kedua kolom data tersebut hampir tidak memiliki hubungan satu sama lain.
##### Berbeda dengan korelasi antara data PetalLengthCm dan PetalWidthCm yang memiliki nilai 0.962757 yang berarti terdapat hubungan yang sangat berkaitan satu sama lain.

# Latihan (15)

Hitung korelasi untuk kolom berikut ini: PetalLengthCm, PetalWidthCm

In [34]:
#Latihan (15)
#Hitung korelasi dataset untuk kolom PetalLengthCm, PetalWidthCm

data[['PetalLengthCm', 'PetalWidthCm']].corr(method='pearson')

PetalLengthCm  PetalWidthCm
PetalLengthCm       1.000000      0.962757
PetalWidthCm        0.962757      1.000000

# Latihan (16)

Method "describe" secara otomatis melakukan komputasi statistik untuk semua continous variable. Secara default "describe" melakukan ignore terhadap variabel bertype objek.

Komputasi statistik yang dilakukan terdiri dari: count, mean, std, min, max, 25%, 75%, max.

Latihan: Gunakan method describe pada dataset yang sudah di load untuk semua continous variabel. (Dataset Iris.csv)

In [22]:
#Latihan (16)
# Penggunaan Metode describe untuk komputasi statistik

data.describe()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
count  150.000000     150.000000    150.000000     150.000000    150.000000
mean    75.500000       5.843333      3.054000       3.758667      1.198667
std     43.445368       0.828066      0.433594       1.764420      0.763161
min      1.000000       4.300000      2.000000       1.000000      0.100000
25%     38.250000       5.100000      2.800000       1.600000      0.300000
50%     75.500000       5.800000      3.000000       4.350000      1.300000
75%    112.750000       6.400000      3.300000       5.100000      1.800000
max    150.000000       7.900000      4.400000       6.900000      2.500000

# Latihan (17)

Gunakan method describe pada dataset yang sudah di load untuk data bertype objek. (Dataset Iris.csv)

In [23]:
#Latihan (17)
#Gunakan method describe pada dataset yang sudah di load untuk data bertype objek

data.describe(include='object')

Species
count           150
unique            3
top     Iris-setosa
freq             50

# Latihan 18

Gunakan method describe pada dataset yang sudah di load untuk semua type data (continous variabel dan type object).

In [10]:
#Latihan (18)
#Gunakan method describe pada dataset yang sudah di load untuk semua type data

data.describe(include='all')

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  \
count   150.000000     150.000000    150.000000     150.000000    150.000000   
unique         NaN            NaN           NaN            NaN           NaN   
top            NaN            NaN           NaN            NaN           NaN   
freq           NaN            NaN           NaN            NaN           NaN   
mean     75.500000       5.843333      3.054000       3.758667      1.198667   
std      43.445368       0.828066      0.433594       1.764420      0.763161   
min       1.000000       4.300000      2.000000       1.000000      0.100000   
25%      38.250000       5.100000      2.800000       1.600000      0.300000   
50%      75.500000       5.800000      3.000000       4.350000      1.300000   
75%     112.750000       6.400000      3.300000       5.100000      1.800000   
max     150.000000       7.900000      4.400000       6.900000      2.500000   

            Species  
count           150  
unique            3  
top     Iris-setosa  
freq             50  
mean            NaN  
std             NaN  
min             NaN  
25%             NaN  
50%             NaN  
75%             NaN  
max             NaN

# Latihan (19)

Hitunglah nilai mean dari dataset.

In [21]:
#Latihan (19)
#Hitung nilai Mean dari dataset

data.mean()

C:\Users\Lenovo\AppData\Local\Temp\ipykernel_9868\3698715605.py:4: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  data.mean()


Id               75.500000
SepalLengthCm     5.843333
SepalWidthCm      3.054000
PetalLengthCm     3.758667
PetalWidthCm      1.198667
dtype: float64

# Latihan (20)

Hitung nilai mean dari dataset untuk kolom PetalLengthCm.

In [7]:
#Latihan (20)
#Hitung nilai Mean untuk kolom PetalLengthCm

petal_length_mean = data['PetalLengthCm'].mean()
petal_length_mean

3.7586666666666693

# Latihan (21)

Carilah nilai minimal dari dataset untuk kolom SepalWidthCm.

In [25]:
#Latihan (21)
#Cari nilai minimal untuk kolom SepalWidthCm

sepal_width_min = data['SepalWidthCm'].min()
sepal_width_min

2.0

# Method Groupby

Method groupby memungkinkan analisis dilakukan secara per kelompok nilai atribut tertentu.

# Latihan (22)

Hitunglah nilai mean dari dataset untuk kolom SepalLengthCm per Species dengan menggunakan metode groupby.

In [17]:
#Latihan (22)
#Hitung nilai mean dari dataset untuk SepalLengthCm per Species dengan metode groupby

sepal_length_mean = data.groupby(['SepalLengthCm', 'Species']).mean()
sepal_length_mean

Id  SepalWidthCm  PetalLengthCm  \
SepalLengthCm Species                                                    
4.3           Iris-setosa       14.000000      3.000000       1.100000   
4.4           Iris-setosa       30.333333      3.033333       1.333333   
4.5           Iris-setosa       42.000000      2.300000       1.300000   
4.6           Iris-setosa       20.500000      3.325000       1.325000   
4.7           Iris-setosa       16.500000      3.200000       1.450000   
4.8           Iris-setosa       25.400000      3.180000       1.580000   
4.9           Iris-setosa       21.250000      3.075000       1.475000   
              Iris-versicolor   58.000000      2.400000       3.300000   
              Iris-virginica   107.000000      2.500000       4.500000   
5.0           Iris-setosa       29.625000      3.362500       1.450000   
              Iris-versicolor   77.500000      2.150000       3.400000   
5.1           Iris-setosa       27.125000      3.600000       1.562500   
              Iris-versicolor   99.000000      2.500000       3.000000   
5.2           Iris-setosa       30.000000      3.666667       1.466667   
              Iris-versicolor   60.000000      2.700000       3.900000   
5.3           Iris-setosa       49.000000      3.700000       1.500000   
5.4           Iris-setosa       17.400000      3.660000       1.540000   
              Iris-versicolor   85.000000      3.000000       4.500000   
5.5           Iris-setosa       35.500000      3.850000       1.350000   
              Iris-versicolor   79.600000      2.440000       3.980000   
5.6           Iris-versicolor   77.200000      2.820000       4.060000   
              Iris-virginica   122.000000      2.800000       4.900000   
5.7           Iris-setosa       17.500000      4.100000       1.600000   
              Iris-versicolor   85.800000      2.820000       4.100000   
              Iris-virginica   114.000000      2.500000       5.000000   
5.8           Iris-setosa       15.000000      4.000000       1.200000   
              Iris-versicolor   81.333333      2.666667       4.000000   
              Iris-virginica   120.000000      2.733333       5.100000   
5.9           Iris-versicolor   66.500000      3.100000       4.500000   
              Iris-virginica   150.000000      3.000000       5.100000   
6.0           Iris-versicolor   78.000000      2.800000       4.525000   
              Iris-virginica   129.500000      2.600000       4.900000   
6.1           Iris-versicolor   75.500000      2.875000       4.500000   
              Iris-virginica   131.500000      2.800000       5.250000   
6.2           Iris-versicolor   83.500000      2.550000       4.400000   
              Iris-virginica   138.000000      3.100000       5.100000   
6.3           Iris-versicolor   72.666667      2.700000       4.666667   
              Iris-virginica   124.500000      2.933333       5.366667   
6.4           Iris-versicolor   63.500000      3.050000       4.400000   
              Iris-virginica   125.600000      2.920000       5.460000   
6.5           Iris-versicolor   55.000000      2.800000       4.600000   
              Iris-virginica   120.250000      3.050000       5.400000   
6.6           Iris-versicolor   67.500000      2.950000       4.500000   
6.7           Iris-versicolor   77.000000      3.066667       4.700000   
              Iris-virginica   133.200000      3.040000       5.600000   
6.8           Iris-versicolor   77.000000      2.800000       4.800000   
              Iris-virginica   128.500000      3.100000       5.700000   
6.9           Iris-versicolor   53.000000      3.100000       4.900000   
              Iris-virginica   134.333333      3.133333       5.400000   
7.0           Iris-versicolor   51.000000      3.200000       4.700000   
7.1           Iris-virginica   103.000000      3.000000       5.900000   
7.2           Iris-virginica   122.000000      3.266667       5.966667   
7.3           Iris-virginica   108.000000  

# Method Value Count

value_counts() menghasilkan frekuensi setiap nilai unik di dalam kolom, dan yang tertinggi count-nya adalah merupakan modus pada kolom tersebut.

# Latihan (23)

Hitunglah frekuensi pada kolom 'Species' dengan menggunakan metode value_counts().

In [11]:
#Latihan (23)
#Hitung frekuensi pada kolom 'Species' dengan menggunakan metode value_counts()

frekuensi_species = data['Species'].value_counts()
frekuensi_species

Iris-setosa        50
Iris-versicolor    50
Iris-virginica     50
Name: Species, dtype: int64

# Latihan (24)

Tampilkan perhitungan frekuensi pada kolom 'Species' dengan menggunakan metode value_counts() dalam bentuk dataframe.

In [26]:
#Latihan (24)
#Perhitungan frekuensi pada kolom 'Species' dengan menggunakan metode value_counts() dalam bentuk dataframe

frekuensi_species = data['Species'].value_counts().to_frame()
frekuensi_species

Species
Iris-setosa           50
Iris-versicolor       50
Iris-virginica        50

# Latihan (25)

Hitunglah frekuensi pada kolom 'PetalLenghCm' dengan menggunakan metode value_counts() dan dalam bentuk dataframe.

In [30]:
#Latihan (25)
# Hitung frekuensi pada kolom 'PetalLenghCm' dengan menggunakan metode value_counts()
frekuensi_petal_length = data['PetalLengthCm'].value_counts().to_frame()
frekuensi_petal_length

PetalLengthCm
1.5             14
1.4             12
5.1              8
4.5              8
1.6              7
1.3              7
5.6              6
4.7              5
4.9              5
4.0              5
4.2              4
5.0              4
4.4              4
4.8              4
1.7              4
3.9              3
4.6              3
5.7              3
4.1              3
5.5              3
6.1              3
5.8              3
3.3              2
5.4              2
6.7              2
5.3              2
5.9              2
6.0              2
1.2              2
4.3              2
1.9              2
3.5              2
5.2              2
3.0              1
1.1              1
3.7              1
3.8              1
6.6              1
6.3              1
1.0              1
6.9              1
3.6              1
6.4              1